lib/xcribe.ex

```diff
@@ -73,38 +71,130 @@ defmodule Xcribe do
     value changes by the format, 'api_blueprint.apib' for Blueprint and
     'app_doc.json' for swagger.
     * `:format` - Format to generate documentation, allowed `:api_blueprint` and
-    `:swagger`. Default `:api_blueprint`.
-    * `:env_var` - Environment variable name for active Xcribe documentation
-    generator. Default is `XCRIBE_ENV`.
+    `:swagger`. Default `:swagger`.
     * `:json_library` - The library to be used for json decode/encode (Jason
     and Poison are supported). The default is the same as `Phoenix` configuration.
     * `:serve` - Enable Xcribe serve mode. Default `false`. See more `Serving doc`
   """
-  use Application
+  require Logger
+
+  alias Xcribe.{
+    ApiBlueprint,
+    CLI.Output,
+    Config,
+    DocException,
+    Recorder,
+    Request,
+    Request.Error,
+    Request.Validator,
+    Swagger,
+    Writter
+  }
 
-  alias Xcribe.CLI.Output
-  alias Xcribe.Config
+  @doc false
+  def document_all_records(override_func \\ nil) do
+    get_records_with_endpoint()
+    |> fetch_config(override_func)
+    |> generate()
+    |> handle_result()
+  end
 
   @doc false
-  def start(_type, _opts) do
-    opts = [strategy: :one_for_one, name: Xcribe.Supervisor]
+  def document(records, config) when is_list(records) do
```

Comment:
This `config` map that is being passed to all sort of functions, would it be better if it was a struct with known fields instead of a regular map?

Reply:
I think so. The next PRs I'll touch the configs again so I'll apply your suggestion. WDYT?